### ISSUE TITLE
False positive for `InterfaceWithOnlyStatics` with sealed interface / nested record subclasses

### ISSUE THREAD

**Issue by kdblocher** [google/error-prone#3389]
The following code leveraging the sealed interface + record pattern (to make sum types) should not violate the [InterfaceWithOnlyStatics](https://errorprone.info/bugpattern/InterfaceWithOnlyStatics) rule as the interface is `sealed`:

```java
public sealed interface Sum {
  record A() implements Sum { }
  record B() implements Sum { }
  // ...anything else
}
```

Note that `permits` is not needed if all implementers are nested types. (The second `record` is not necessary, but illustrates the point.)

**Comment by graememorgan** [comment]
We fixed this at some point by excluding `sealed` interfaces.
